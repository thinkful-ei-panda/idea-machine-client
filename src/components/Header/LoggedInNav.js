import React from 'react';
import {NavLink} from 'react-router-dom';
import TokenService from '../../services/token-service';
import './Nav.css';

class LoggedInNav extends React.Component {
  handleLogoutClick = () => {
    TokenService.clearAuthToken();
    this.setState({});
  }

  render() {
    return (
      <nav className='Nav'>
        <NavLink to='/home'>
          <div>Home</div>
        </NavLink>
        <NavLink to='/my-ideas'>
          <div>My Ideas</div>
        </NavLink>
        <NavLink to='/tracked-ideas'>
          <div>Following</div>
        </NavLink>        
        <NavLink onClick={this.props.handleLogout} to='/login' >
          <div>Logout</div>
        </NavLink>
      </nav>
    )
  }  
}

export default LoggedInNav