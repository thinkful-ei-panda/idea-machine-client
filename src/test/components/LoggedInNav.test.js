import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import LoggedInNav from '../../components/Header/LoggedInNav';
describe(`LoggedInNav Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <LoggedInNav />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <LoggedInNav />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})