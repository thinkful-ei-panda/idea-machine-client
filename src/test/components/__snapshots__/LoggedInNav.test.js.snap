// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedInNav Component Snapshot test Renders the UI as expected 1`] = `
<nav
  className="Nav"
>
  <a
    aria-current={null}
    href="/home"
    onClick={[Function]}
  >
    <div>
      Home
    </div>
  </a>
  <a
    aria-current={null}
    href="/my-ideas"
    onClick={[Function]}
  >
    <div>
      My Ideas
    </div>
  </a>
  <a
    aria-current={null}
    href="/tracked-ideas"
    onClick={[Function]}
  >
    <div>
      Following
    </div>
  </a>
  <a
    aria-current={null}
    href="/login"
    onClick={[Function]}
  >
    <div>
      Logout
    </div>
  </a>
</nav>
`;
